### ISSUE TITLE
Version 20.0.1 Installation Error

### ISSUE THREAD

**Issue by CoreTrust** [ArtificialOwl/files_fulltextsearch_tesseract#41]
Nextcloud displays the following error message during installation:

Client error: `GET https://github.com/daita/files_fulltextsearch_tesseract/releases/download/v20.0.1/files_fulltextsearch_tesseract-20.0.1.tar.gz` resulted in a `404 Not Found` response: Not Found.

Does anyone have an idea?

**Comment by ann0see** [comment]
https://github.com/ArtificialOwl/files_fulltextsearch_tesseract/issues/38 seems to be related?
Why is @daita's fork linked? Is this the official app store release?

Maybe we can download https://github.com/ArtificialOwl/files_fulltextsearch_tesseract/releases/tag/v21.0.0 and install it manually.

**Comment by andrewborell** [comment]
> Maybe we can download https://github.com/ArtificialOwl/files_fulltextsearch_tesseract/releases/tag/v21.0.0 and install it manually.

I don't think @daita has anything to do with this. The releases are still there but the link in the nextcloud store app to retrieve the tarball is broken. This could very well be a change in github I suppose but I don't use github for much. That aside, I would not recommend installing the v21 release to Nextcloud v20, but yes, you can install it manually.  

If you don't know your user then take note with ls -lah on your nextcloud site to see who owns it all. If your www folders happen to show the owner is root then immediately stop what you are doing, log out of your computer, and never log back in again.  Adjust the commands below for your www user and site path as needed. Also if you want to pull the artificial own release then change the url. 

`cd /tmp`
`wget https://github.com/daita/files_fulltextsearch_tesseract/archive/refs/tags/v20.0.1.tar.gz`
`tar -xf v20.0.1.tar.gz`
`mkdir files_fulltextsearch_tesseract_backup`
`mv /var/www/your_site.com/nextcloud/apps/files_fulltextsearch_tesseract/* /tmp/files_fulltextsearch_tesseract_backup/`
`mv /tmp/files_fulltextsearch_tesseract-20.0.1/* /var/www/your_site.com/nextcloud/apps/files_fulltextsearch_tesseract/`
`chown -R www-data:www-data /var/www/your_site.com/nextcloud/apps/files_fulltextsearch_tesseract-20.0.1`
`sudo -H -u www-data php /var/www/your_site.com/nextcloud/occ app:list`
`sudo -H -u www-data php /var/www/your_site.com/nextcloud/occ app:enable files_fulltextsearch_tesseract`

Clean up your /tmp/ directory at your discretion. This information is provided at your own risk.

**Comment by ann0see** [comment]
Yeah, installed it quite similar to your guide above. Can the link on the App Store be updated by someone externally?

**Comment by p3pp0** [comment]
not working on NC22: any idea?

`PHP Fatal error:  require_once(): Failed opening required '/var/www/html/apps/files_fulltextsearch_tesseract/lib/AppInfo/../../vendor/autoload.php' (include_path='/var/www/html/3rdparty/pear/archive_tar:/var/www/html/3rdparty/pear/console_getopt:/var/www/html/3rdparty/pear/pear-core-minimal/src:/var/www/html/3rdparty/pear/pear_exception:/var/www/html/apps') in /var/www/html/apps/files_fulltextsearch_tesseract/lib/AppInfo/Application.php on line 45
`

**Comment by theLSB** [comment]
@andrewborell thans for your extended comment.
Indeed, the URL to the release in the nextcloud store is wrong:
The store is reffering to:
https://github.com/daita/files_fulltextsearch_tesseract/releases/download/v20.0.1/files_fulltextsearch_tesseract-20.0.1.tar.gz
While the actual url is:
https://github.com/daita/files_fulltextsearch_tesseract/archive/refs/tags/v20.0.1.tar.gz 

I too am running a local private nextcloud instance so I have full control, yet I could not get the app to work - following your instructions. 
After putting the folder in place, the WebUI shows the app as needed to be downloaded + installed, AND - that it can be removed.
Running occ app:list lists it under the disabled apps.

When trying to enable it using occ I get:
```
# sudo -u www-data ../occ app:enable files_fulltextsearch_tesseract-20.0.1
App not found
```
even though it is listed under the disabled apps.
It is interesting since when I do app:check-code on it I get:
```
# sudo -u www-data ../occ app:check-code files_fulltextsearch_tesseract-20.0.1
App is compliant - awesome job!
```
So I am not sure what the issue is.
Tesseract is installed, and ownership and permissions seem to check too....

If you have ideas what I can try I will appreciate it very much.

**Comment by andrewborell** [comment]
@theLSB, I have not spent much time looking at it since then. When I enabled that app it caused the site to become inaccessible until I backed out the changes, so this has become an after-hours topic for me moving forward, and I haven't put anything into it since my last post. I will maybe take a look tonight to see if I can get it working ( or partially working ) again.  I wonder if your issue is you don't have the right user/group on the files you extracted?? i.e. my chown -R line from above needs to be different for your setup.  Just a shot in the dark without knowing more about your situation.

**Comment by andrewborell** [comment]
Took a peek at my install. Something is going wrong with the build process causing my site to crash. The composer file looks okay in itself, but I can clearly see the vendor directory is not being created as well as a bunch of other things. I didn't have much time yesterday because the LE intermediate cert expiration caused some issues with onlyoffice integration but I will figure this out tonight. I kind of miss the OCR for new documents.
